### ISSUE TITLE
`torch.nanmean` only expects input to have floating point

### ISSUE THREAD

**Issue by TestSomething22** [pytorch/pytorch#71472]
### 🐛 Describe the bug

Based on the documentation of `torch.nanmean`, it is identical to `torch.mean()` when there are no NaN values in the input tensor. However, it only expects input to have floating point and complex tensor can have NaN.

```python
import torch
input= torch.rand([6, 5, 5], dtype=torch.complex128)
print(torch.mean(input))
# tensor(0.5126+0.5373j, dtype=torch.complex128)
torch.nanmean(input)
# RuntimeError: nanmean(): expected input to have floating point dtype but got ComplexDouble
```

```python
import torch
input = torch.rand([2, 2], dtype=torch.complex64)
input[0] = torch.nan
print(torch.mean(input))
# tensor(nan+nanj)
torch.nanmean(input)
# RuntimeError: nanmean(): expected input to have floating point dtype but got ComplexFloat
```

### Versions

pytorch: 1.10.1

**Comment by Lezcano** [comment]
I don't think this is so much a bug as a feature request extending nanmean to complex numbers. Now, it's not clear what should be done with numbers such as `1+NaNj` or `NaN+2j`. It would be good to look at other nan functions in PyTorch to be consistent with them. We should also look at other libraries see how they do this

**Comment by mruberry** [comment]
We actually did decide how to define complex NaN, see here: https://pytorch.org/docs/master/generated/torch.isnan.html?highlight=isnan#torch.isnan. But yes, we just don't support complex nanmean, which is a feature request, and we should add support for it if users want it. 

We would take a PR for this. cc @kshitij12345 who's thinking about extending complex support for PyTorch 1.12.
